org/all-prs.js

```diff
@@ -9,106 +9,126 @@
-  reject, split, startsWith,
+  all,
+  any,
+  both,
+  complement,
+  compose,
+  contains,
+  filter,
+  flip,
+  head,
+  join,
+  last,
+  map,
+  path,
+  pipe,
+  prop,
+  reject,
+  split,
+  startsWith,
+  test,
+  toPairs,
 } = require('ramda');
 
-// # Markdown utils
-const mdList = items => isEmpty(items) ? '' : `- ${join('\n- ', items)}`;
-exports.mdList = mdList;
+const isAscii = exports.isAscii = str => Buffer.from(str, 'utf8').toString('ascii') === str;
+const mdList = exports.mdList = items => join('\n', items.map(t => `- ${t}`));
 
 //eslint-disable-next-line no-console
 const ok = test => console.info(`${test} OK`);
 
 // # Commits
 // Rules related to commit descriptions and body
 
-// TODO: consider using https://github.com/marionebl/commitlint
-const commitTypes = [
-  'feat:',
-  'fix:',
-  'test:',
-  'ci:',
-  'chore:',
-  'docs:',
-  'refactor:',
-  'style:',
-  'perf:',
-  'revert:',
+const verbs = readFileSync(resolve(__dirname, '../data/verbs.txt'), 'utf8').split('\n');
+const approvedVerbs = [
+  'Add',
+  'Remove',
+  'Fix',
+  'Test',
+  'Document',
+  'Refactor',
+  'Style',
+  'Revert',
+  'Update',
+  'Configure',
+  'Deprecate',
+  'Correct',
+  'Improve',
+  'Initialise',
+  'Merge',
+  'Release',
 ];
 
 const linkForCommit = commit => `[${commit.sha.slice(0, 6)}](${commit.url})`;
-
 const commitMsg = prop('message');
-
-const startsWithI = needle => haystack => haystack.toLowerCase().startsWith(needle);
-const unknownPrefix = compose(none(Boolean), ap(map(startsWithI, commitTypes)), of);
-const commitsWithUnkownType =
-  filter(compose(unknownPrefix, commitMsg));
-
-const excludeMergeCommits =
-  reject(compose(startsWith('Merge'), commitMsg));
-
-const hasLinesOver = n => compose(
-  any(line => line.length > n),
-  split('\n'),
-);
-const commitsWithLinesOver = n =>
-  filter(compose(hasLinesOver(n), commitMsg));
-
-const reportCommits = (reporter, messageFn) => compose(
-  reporter,
-  mdList,
-  map(messageFn),
-);
-
-const maxCommitLineLength = 100;
-/**
- * Fails when any commit except Merge commits have a line in their body that exceeds 100 characters
- */
-exports.lineLength = () => {
-  const offendingCommits = compose(
-    commitsWithLinesOver(maxCommitLineLength),
-    excludeMergeCommits
-  )(danger.git.commits);
-
-  if (!isEmpty(offendingCommits)) {
-    reportCommits(fail, commit =>
-      `Commit ${linkForCommit(commit)} has lines with over ${maxCommitLineLength} chars.`
-    )(offendingCommits);
-  } else {
-    ok('lineLength');
+const excludeMergeCommits = reject(compose(startsWith('Merge'), commitMsg));
+
+exports.tests = {
+
+  commitApprovedVerb: {
+    citical: false,
+    test: pipe(
+      path(['git', 'commits']),
+      reject(pipe(commitMsg, split(' '), head, flip(contains)(approvedVerbs))),
+      map(linkForCommit),
+      map(commit => (
+        `Message for commit ${commit} should start with one of: ${approvedVerbs.join(', ')}.`
```

Comment:
Not much context for the user here, maybe `Message for commit ${commit} has uncommon verb, consider using ... instead.`